sky/cli.py

```diff
@@ -3180,19 +3180,26 @@ def _output():
                 yield '\n\n'
             yield from tpu_table.get_string()
 
-            # Other GPUs
-            if show_all:
+            # Handle Other GPUs
+            if show_all or cloud is not None:
                 yield '\n\n'
                 for gpu, qty in sorted(result.items()):
                     other_table.add_row([gpu, _list_to_str(qty)])
                 yield from other_table.get_string()
                 yield '\n\n'
-            else:
+
+            # Handle hints and messages
+            if not show_all and cloud is None:
```

Comment:
We may need to do some special handling here. Refer to the `master` branch output and output in this branch below.

1. We want to retain the hint at the end, but exclude the `(including non-common ones)` bit. I.e., we should print a hint like `Hint: use -a/--all to see all accelerators and pricing.` when `sky show-gpus --cloud xyz` is run, and the current hint `Hint: use -a/--all to see all accelerators (including non-common ones) and pricing.` when `sky show-gpus` is run.
2. There's some extra empty lines at the end of this branch's output.

Master:
```
(base) ➜  ~ sky show-gpus --cloud aws
COMMON_GPU  AVAILABLE_QUANTITIES
A10G        1, 4, 8
A100        8
A100-80GB   8
H100        8
K80         1, 8, 16
L4          1, 4, 8
M60         1, 2, 4
T4          1, 4, 8
V100        1, 4, 8
V100-32GB   8

Hint: use -a/--all to see all accelerators (including non-common ones) and pricing.
```

This branch:
```
(base) ➜  ~ sky show-gpus --cloud aws
COMMON_GPU  AVAILABLE_QUANTITIES
A10G        1, 4, 8
A100        8
A100-80GB   8
H100        8
K80         1, 8, 16
L4          1, 4, 8
M60         1, 2, 4
T4          1, 4, 8
V100        1, 4, 8
V100-32GB   8

OTHER_GPU        AVAILABLE_QUANTITIES
Gaudi HL-205     8
L40S             1, 4, 8
Radeon Pro V520  1, 2, 4
T4g              1, 2


```